BAZProject/DetailsMovies/DetailsView.swift

```diff
@@ -6,7 +6,165 @@
 //
 
 import UIKit
+import MapKit
+import CoreLocation
 
-class DetailsView: UIViewController {
+class DetailsView: UIViewController, CLLocationManagerDelegate {
     
+    //MARK: Oulets
+    
+    var viewModel: DetailsViewModel
+    let locationManager = CLLocationManager()
+    @IBOutlet weak var imageMovie: UIImageView!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var overviewText: UILabel!
+    @IBOutlet weak var voteAverage: UILabel!
+    @IBOutlet weak var actorsCV: UICollectionView!
+    @IBOutlet weak var similarCV: UICollectionView!
+    @IBOutlet weak var recomendationsCV: UICollectionView!
+    @IBOutlet weak var favoriteButton: UIButton!
+    @IBOutlet weak var mapView: MKMapView!
+    
+    init(viewModel: DetailsViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: "DetailsView", bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setView()
+        setmap()
+        viewModel.bindMovie { [weak self] in
+            DispatchQueue.main.async {
+                self?.actorsCV.reloadData()
+                self?.similarCV.reloadData()
+                self?.recomendationsCV.reloadData()
+            }
+        }
+    }
+
+    /// Registro de los nibs para el uso de todos los CollectionViews
+    func registerNibs() {
+        let nib = UINib(nibName: "DetailsCollectionViewCell", bundle: nil)
+        actorsCV.register(nib, forCellWithReuseIdentifier: "cell")
+        similarCV.register(nib, forCellWithReuseIdentifier: "cell")
+        recomendationsCV.register(nib , forCellWithReuseIdentifier: "cell")
+    }
+    
+    /// Se setea todo lo necesario para cargar la vista
+    func setView() {
+        registerNibs()
+        getAllInfo()
+        setFavoriteButton()
+        titleLabel.text = viewModel.getTitle()
+        navigationItem.title = viewModel.getTitle()
+        imageMovie.loadImage(url: URL(string: "https://image.tmdb.org/t/p/w500/\(viewModel.getPathImage() ?? "")")!)
+        overviewText.text = viewModel.getOverview()
+        voteAverage.text = viewModel.getRating()
+    }
+    
+    ///Consulta todos los servicios
+    func getAllInfo() {
+        viewModel.getCastMovie()
+        viewModel.getSimilarMovies()
+        viewModel.getRecommendationMovies()
+    }
+    /// Verifica si la pelicula esta en favoritos y cambia el titulo
+    func setFavoriteButton() {
+        viewModel.isMovieFavorite() ? favoriteButton.setImage(UIImage(systemName: "heart.fill"), for: .normal) : favoriteButton.setImage(UIImage(systemName: "heart"), for: .normal)
```

Comment:
Podrías crear también un enum que se encargue de guardar los iconos.
Algo así:

```
/// Enum used to store the icons used through the app.
enum Icon {
    
    static let heart: UIImage = UIImage(systemName: "heart")
    static let heartFill: UIImage = UIImage(systemName: "heart.fill")
}
```


```suggestion
        let icon = viewModel.isMovieFavorite() ? Icon.heartFill : Icon.heart
        favoriteButton.setImage(icon, for: .normal)
```